vita3k/gui/src/imgui_impl_sdl.cpp

```diff
@@ -120,6 +120,13 @@ IMGUI_API void ImGui_ImplSdl_NewFrame(ImGui_State *state) {
     int display_w, display_h;
     SDL_GetWindowSize(state->window, &w, &h);
     ImGui_ImplSdl_GetDrawableSize(state, display_w, display_h);
+#if __APPLE__
+    // On macOS, if HiDPI is enabled(SDL_WINDOW_ALLOW_HIGHDPI) SDL will create a window that size is 960x544 points.
+    // But it is actually 1920x1088 pixels. Since SDL_GetWindowSize returns 960x544 points, we need to multiply by 2.
+    w *= 2;
+    h *= 2;
+#endif
+
```

Comment:
SDL_GetWindowSize returns the screen dimensions in coordinates while ImGui_ImplSdl_GetDrawableSize returns the same screen dimension in pixel, this is done because of hidpi scaling. What you are doing is making ImGui believe the dpi scaling is 1.0 which sounds kind of counter-intuitive (and also forces you to implement workarounds like the one below for SDL_GetMouseState)

Reply:
But from my windows pc with 150% scale, EmuEnvState's dpi_scale is 1.5 but it's w and display_w is same(960) and h and display is also same(544) so ImGui's DisplayFramebufferScale is set to (1, 1). 

Also as screenshot shows it was rendering at real pixel. So I was thinking HiDPI is correctly working.